common/docdiff/src/main.rs

```diff
@@ -294,38 +301,36 @@ fn index_file(path: &Path) -> Result<Vec<Entry>> {
         }
     }
 
-    for trait_impl in document.select(sel!("#trait-implementations-list details")) {
-        let types = trait_impl
-            .select(sel!(".impl-items .trait-impl.type:not(.hidden)"))
-            .map(|t| (t, Kind::Type));
-        let constants = trait_impl
-            .select(sel!(
-                ".impl-items .trait-impl.associatedconstant:not(.hidden)"
-            ))
-            .map(|t| (t, Kind::Constant));
-
-        let items = types.chain(constants);
-
-        let signature = el_to_string(trait_impl.select(sel!(".impl")).next().unwrap());
-
+    for trait_impl in document.select(sel!("#trait-implementations-list details .impl")) {
+        let signature = el_to_string(trait_impl);
         results.push(Entry {
             kind: Kind::Trait,
             signature: signature.clone(),
             context: context.clone(),
         });
 
-        let context = Arc::new(Fqn {
-            kind: Kind::Trait,
-            path: signature,
-        });
-
-        for (item, kind) in items {
-            let signature = el_to_string(item);
-            results.push(Entry {
-                kind,
-                signature,
-                context: context.clone(),
-            })
+        for trait_impl in document.select(sel!("#trait-implementations-list details")) {
+            let context = Arc::new(Fqn {
```

Comment:
Why is this getting created for each iteration? You should be able to move it up a level.

Reply:
https://github.com/aws/s2n-quic/pull/1433/commits/8e43211908af46c1b3436a3e39defc856c15213a